src/WebJobs.Extensions.Http/ClaimsPrincipalBindingProvider.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Security.Claims;
+using System.Security.Principal;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Protocols;
+using static Microsoft.Azure.WebJobs.Extensions.Http.HttpTriggerAttributeBindingProvider.HttpTriggerBinding;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Http
+{
+    /// <summary>
+    /// This provider provides a binding to Type <see cref="ClaimsPrincipal"/>.
+    /// </summary>
+    internal class ClaimsPrincipalBindingProvider : IBindingProvider
+    {
+        private static readonly Task<IBinding> NullBinding = Task.FromResult<IBinding>(null);
+
+        public Task<IBinding> TryCreateAsync(BindingProviderContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if (context.Parameter.ParameterType != typeof(ClaimsPrincipal))
+            {
+                return NullBinding;
+            }
+
+            return Task.FromResult<IBinding>(new ClaimsIdentityBinding(context.Parameter));
```

Comment:
Since this is named ClaimsPrincipalBindingProvider, shouldn't it be providing a ClaimsPrincipalBinding rather than a ClaimsIdentityBinding? I.e. probably need to rename your private class?